decidim-core/spec/lib/form_builder_spec.rb

```diff
@@ -378,5 +379,88 @@ def validate_presence
         end
       end
     end
+
+    describe "upload" do
+      let(:present?) { false }
+      let(:content_type) { nil }
+      let(:filename) { "my_image.jpg" }
+      let(:url) { "/some/file/path/#{filename}" }
+      let(:file) do
+        double(
+          url: url,
+          present?: present?,
+          content_type: content_type,
+          file: double(
+            filename: filename
+          )
+        )
+      end
+      let(:optional) { true }
+      let(:attributes) do
+        {
+          optional: optional
+        }
+      end
+      let(:output) do
+        builder.upload :image, attributes
+      end
+
+      before do
+        allow(resource).to receive(:image).and_return(file)
+      end
+
+      it "sets the form as multipart" do
+        output
+        expect(builder.multipart).to be_truthy
+      end
+
+      it "renders a file_field" do
+        expect(parsed.css('input[type="file"]').first).to be
+      end
+
+      context "when it is an image" do
+        it "renders an image with the current file url" do
+          expect(parsed.css('img[src="' + url + '"]').first).to be
+        end
+      end
+
+      context "when it is not an image" do
+        let(:filename) { "my_file.pdf" }
+
+        it "doesn't render an image tag" do
+          expect(parsed.css('img[src="' + url + '"]').first).not_to be
+        end
+      end
+
+      context "when the file is not present" do
+        it "renders the 'Default image' label" do
+          expect(output).to include("Default image")
+        end
+      end
+
+      context "when the file is present" do
+        let(:present?) { true }
+
+        it "renders the 'Current image' label" do
+          expect(output).to include("Current image")
+        end
+
+        it "renders a link to the current file url" do
```

Comment:
I don't think this provides much value and looks quite weird on the interface level. Maybe the link should be on the image itself.